Architecture diagram (#111)
SW709E19/Report: report/PartDesign/ChapterArchitecture/ArchitectureDiagram.tex

The sensor device is the main foundation of the entire system.
Without the sensor device no data would be collected and thus no predictive maintenance can be conducted.
It is thus necessary for the sensor device to be both reliable and ridged enough to work in harsh conditions.

\section{Gateway}

A gateway is the local link between the sensor device and the cloud.
The gateway is thus used to manage the connection between the two core components ie. the sensor devices and the cloud.

\subsection{Tasks}

The gateway has a number of tasks, it has to receive messages from the sensor devices and relay them to the cloud.
Furthermore it should also relay messages form the cloud out to the sensor devices with new configurations.

\subsection{Responsibilities}

The gateway is responsible for handling the authentication to the cloud within the publish/subscribe method.
The gateway is responsible for updating devices as soon as they come out of the sleep cycle if an updated has been pushed through the cloud.

\section{Backend}

The backend is on \autoref{Fig:ArchitectureDiagram} denoted as the "cloud" for a more generic approach the is agnostic of different cloud providers.

> The backend should be interchangeable with many different providers and types as best suited for the intended implementation.

Reviewer: ```suggestion
The backend should be interchangeable with different providers to allow for the provider that is best suited for the intended implementation.
```

Reviewer: Der er ingen grund til at sige "many different provideres" bare at det ikke skal være bundet til en provider

\subsection{Tasks}

The backend has a number of tasks such as receiving messages from the gateways and handle them appropriately.

> It also has to send messages to the devices with new configurations of their settings.

Reviewer: ```suggestion
It also has to handle the communication of new configurations to the devices.
```

The backend is also responsible for authentication of all devices, both gateways and sensor devices.

\subsection{Responsibilities}

> The backend is responsible for that once a message is received it is saved for later analysis. 

Reviewer: ```suggestion
The backend is responsible for saving received messages for later analysis.

```

> The backend also has to authenticate each device and implement different levels of authentication such that a sensor device can only publish new information and delete existing measurements.

Reviewer: ```suggestion
The backend also has to authenticate each device and implement different permissions such that a sensor device can only publish new information and delete existing measurements.
```

\section{Storage}

> The storage is a generic term for any type of persistent storage and can thus very depending on individual needs.

Reviewer: ```suggestion
The storage is a generic term for any type of persistent storage and can thus vary depending on individual needs.
```

\subsection{Tasks}

> The storage has to store the data collected by the system, it also has to make the data available for the analyst upon request.

Reviewer: ```suggestion
The storage has to store the data collected by the system and make the data available for the analyst upon request.
```

\subsection{Responsibilities}

> Once the data is stored in the storage it should be kept safe, also in case of a service failure or crash.

Reviewer: ```suggestion
Once the data is stored in the storage it should be kept safe and redundant in the case of a service failure.
```

\section{Frontend}

The frontend is meant to be a dashboard where the maintenance personnel can see the status of their factory's sensor network.

\subsection{Tasks}

The dashboard can be said to have one responsibility, allowing for the maintenance personnel to see the status of their factory. 
The status should consist of a number of data points, such as an overview of connected devices, their last heartbeat and their battery status.

\subsection{Responsibilities}

> With regards to responsibilities the dashboard does not have any since it is just a visualization of data stored else where.

Reviewer: VI kan vel skrive noget om at "responsibilities" skal være at det skal være bruger venligt præsenteret og skal vise akkurat information osv
